File: CBLClient/Apps/JavaRequestHandlers/src/main/java/com/couchbase/mobiletestkit/javacommon/RequestHandler/DatabaseConfigurationRequestHandler.java
package com.couchbase.mobiletestkit.javacommon.RequestHandler;


import com.couchbase.CouchbaseLiteServ.TestServerApp;
import com.couchbase.CouchbaseLiteServ.util.Log;
import com.couchbase.lite.DatabaseConfiguration;
import com.couchbase.lite.EncryptionKey;
import com.couchbase.mobiletestkit.javacommon.Args;


public class DatabaseConfigurationRequestHandler {
    private static final String TAG = "DATABASE_CONFIG";

    public DatabaseConfiguration configure(Args args) {
        String directory = args.getString("directory");
        if (directory == null) {
            directory = TestServerApp.getApp().getFilesDir().getAbsolutePath();
            Log.i(TAG, "No directory is set, now point to " + directory);
        }
        Log.i(TAG, "DatabaseConfiguration_configure directory=" + directory);
        DatabaseConfiguration config = new DatabaseConfiguration();
        config.setDirectory(directory);

        EncryptionKey encryptionKey;
        String password = args.getString("password");
        if (password != null) {
            encryptionKey = new EncryptionKey(password);
            config.setEncryptionKey(encryptionKey);
        }
        return config;
    }

    public String getDirectory(Args args) {
        DatabaseConfiguration config = args.get("config", DatabaseConfiguration.class);
        return config.getDirectory();
    }

    public EncryptionKey getEncryptionKey(Args args) {
        DatabaseConfiguration config = args.get("config", DatabaseConfiguration.class);
        return config.getEncryptionKey();
    }

    public DatabaseConfiguration setDirectory(Args args) {
        DatabaseConfiguration config = args.get("config", DatabaseConfiguration.class);
        String directory = args.getString("directory");
        return config.setDirectory(directory);
    }

    public DatabaseConfiguration setEncryptionKey(Args args) {
        DatabaseConfiguration config = args.get("config", DatabaseConfiguration.class);
        String password = args.getString("password");
        EncryptionKey encryptionKey = new EncryptionKey(password);
        return config.setEncryptionKey(encryptionKey);
    }
}

File: CBLClient/Apps/JavaRequestHandlersKotlin/src/main/java/com/couchbase/mobiletestkit/javacommon/RequestHandler/DatabaseConfigurationRequestHandler.kt
package com.couchbase.mobiletestkit.javacommon.RequestHandler

import com.couchbase.lite.DatabaseConfiguration
import com.couchbase.lite.EncryptionKey
import com.couchbase.lite.create
import com.couchbase.mobiletestkit.javacommon.Args
import com.couchbase.mobiletestkit.javacommon.RequestHandlerDispatcher
import com.couchbase.mobiletestkit.javacommon.util.Log

class DatabaseConfigurationRequestHandler {
    fun configure(args: Args): DatabaseConfiguration {
        var directory = args.get<String>("directory")
        if (directory == null) {
            directory = RequestHandlerDispatcher.context.filesDir.absolutePath
            Log.i(
                TAG,
                "No directory is set, now point to $directory"
            )
        }
        Log.i(
            TAG,
            "DatabaseConfiguration_configure directory=$directory"
        )
        val config = DatabaseConfiguration()
        config.create(directory!!)
        val encryptionKey: EncryptionKey
        val password = args.get<String>("password")
        if (password != null) {
            encryptionKey = EncryptionKey(password)
            config.setEncryptionKey(encryptionKey)
        }
        return config
    }

    /*public ConflictResolver getConflictResolver(Args args){
        DatabaseConfiguration config = args.get("config");
        return config.getConflictResolver();
    }*/
    fun getDirectory(args: Args): String {
        val config = args.get<DatabaseConfiguration>("config")
        return config.directory
    }

    fun getEncryptionKey(args: Args): EncryptionKey? {
        val config = args.get<DatabaseConfiguration>("config")
        return config.encryptionKey
    }

    /*public DatabaseConfiguration setConflictResolver(Args args){
        DatabaseConfiguration config = args.get("config");
        ConflictResolver conflictResolver = args.get("conflictResolver");
        return config.setConflictResolver(conflictResolver);
    }*/
    fun setDirectory(args: Args): DatabaseConfiguration {
        val config = args.get<DatabaseConfiguration>("config")
        val directory = args.get<String>("directory")
        return config.setDirectory(directory)
    }

    fun setEncryptionKey(args: Args): DatabaseConfiguration {
        val config = args.get<DatabaseConfiguration>("config")
        val password = args.get<String>("password")
        val encryptionKey = EncryptionKey(password)
        return config.setEncryptionKey(encryptionKey)
    }

    companion object {
        private const val TAG = "DATABASE_CONFIG"
    }
}